web/src/ResourceGroupsContext.tsx

```diff
@@ -0,0 +1,81 @@
+import { createContext, PropsWithChildren, useContext } from "react"
+import { AnalyticsAction, AnalyticsType, incr } from "./analytics"
+import { usePersistentState } from "./LocalStorage"
+
+export type GroupState = { expanded: boolean }
+
+export type GroupsState = {
+  [key: string]: GroupState
+}
+
+type ResourceGroupsContext = {
+  groups: GroupsState
+  getGroup: (groupLabel: string) => GroupState
+  toggleGroupExpanded: (groupLabel: string, page: AnalyticsType) => void
+}
+
+export const DEFAULT_EXPANDED_STATE = true
+export const DEFAULT_GROUP_STATE: GroupState = {
+  expanded: DEFAULT_EXPANDED_STATE,
+}
+
+const resourceGroupsContext = createContext<ResourceGroupsContext>({
+  groups: {},
+  toggleGroupExpanded: () => {
+    console.warn("Resource group context is not set.")
+  },
+  getGroup: () => {
+    console.warn("Resource group context is not set.")
+    return { ...DEFAULT_GROUP_STATE }
+  },
+})
```

Comment:
@nicks, i addressed [your feedback](https://github.com/tilt-dev/tilt/pull/4845#discussion_r687157169) here by changing the shape of the resource groups state from 
```
state: { labelName: boolean }
```
 to 
```
state: { labelName: { expanded: boolean } }
```
so it's clearer and more extensible if there's other state we want to track for label groups. the method names are also renamed to be clearer. (it may be a little awkward that there's a `getGroup` but not a `setGroup` where you can pass in the desired group state? i figured a `toggle` method is more useful for the implementation now, but i can refactor that.)

Reply:
:+1: